### ISSUE TITLE
Why do we need Context as a struct/class?

### ISSUE THREAD

**Issue by kechan** [tensorflow/swift#144]
Pls let me know if this is right place to post this sort of comment.

Refer to:

```
let trainingContext = Context(learningPhase: .training)
let untrainedLogits = model.applied(to: firstTrainFeatures, in: trainingContext)
```

Wondering why we need a whole class or struct "Context". It looks like the single piece of external info is ".training"? So why can't this be an enum. I may be asking something naive here though. But this looks simpler:

```
let untrainedLogits = model.applied(to: firstTrainFeatures, withContext: .training)
```

**Comment by rxwei** [comment]
This is a design question we are actively exploring. Ideally we don’t want the user to pass this argument at all in their implementation of ‘applied(to:in:)’ method because it’s just boilerplatey. Follow our mailing list and we’ll have many discussions about this topic soon!

**Comment by kechan** [comment]
Thanks. Can you pls kindly point out what mailing list this is, I can join it. 

If there's someway to infer the context without explicit needing to pass, then that will be great.

**Comment by dan-zheng** [comment]
Hi @kechan,

Here's the [Swift for TensorFlow mailing list](https://groups.google.com/a/tensorflow.org/forum/#!forum/swift), and here's the original [thread](https://groups.google.com/a/tensorflow.org/forum/#!searchin/swift/edward|sort:date/swift/PtiIr83Xfgk/Tiige_LLAAAJ)/[proposal PR](https://github.com/tensorflow/swift/pull/115) introducing the `Context` class.

You can search for "context" to find previous discussion about the `Context` design - more discussions coming soon!

**Comment by rxwei** [comment]
The `in: context` parameter has been removed in the latest design. The new design is now easier to use when combined with `Layer` being a callable object:
```swift
let untrainedLogits = model(firstTrainFeatures)
```

This will be released in Swift for TensorFlow v0.3.
